src/postings/recorder.rs

```diff
@@ -125,18 +117,22 @@ impl Recorder for DocIdRecorder {
         let (buffer, doc_ids) = buffer_lender.lend_all();
         self.stack.read_to_end(arena, buffer);
         // TODO avoid reading twice.
+        let mut prev_doc = 0;
         if let Some(doc_id_map) = doc_id_map {
-            doc_ids.extend(
-                VInt32Reader::new(&buffer[..])
-                    .map(|old_doc_id| doc_id_map.get_new_doc_id(old_doc_id)),
-            );
+            doc_ids.extend(VInt32Reader::new(&buffer[..]).map(|delta_doc_id| {
```

Comment:
Usage of `prev_doc` here looks like [`scan`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.scan) instead of `map` could make this more obvious.